serializable.ts

```diff
@@ -4,8 +4,21 @@ import { SerializePropertyOptionsMap } from "./serialize_property_options_map.ts
 import { defaultToJson } from "./to_json/default.ts";
 import { recursiveToJson } from "./to_json/recursive.ts";
 
+/** to be implemented by external authors on their models  */
+export declare interface TransformKey {
+  /** a function that will be called against
+   * every property key transforming the key
+   * with the provided function
+   */
+  tsTransformKey(key: string): string;
+}
+
 /** Adds methods for serialization */
 export abstract class Serializable {
+  /** the default transform functionality */
+  public tsTransformKey?(key: string): string {
```

Comment:
I am a little sad we can't make this `static` due to typescript limitations, at least for now https://github.com/microsoft/TypeScript/issues/33892